index.js

```diff
@@ -238,22 +238,29 @@ ShareDbMongo.prototype.close = function(callback) {
 
 ShareDbMongo.prototype.commit = function(collectionName, id, op, snapshot, options, callback) {
   var self = this;
-  var request = createRequestForMiddleware(options, collectionName, op);
+  var request = createRequestForMiddleware(null, options, collectionName, op);
   this._writeOp(collectionName, id, op, snapshot, function(err, result) {
     if (err) return callback(err);
     var opId = result.insertedId;
     self._writeSnapshot(request, id, snapshot, opId, function(err, succeeded) {
       if (succeeded) return callback(err, succeeded);
       // Cleanup unsuccessful op if snapshot write failed. This is not
-      // neccessary for data correctness, but it gets rid of clutter
+      // necessary for data correctness, but it gets rid of clutter
       self._deleteOp(request.collectionName, opId, function(removeErr) {
         callback(err || removeErr, succeeded);
       });
     });
   });
 };
 
-function createRequestForMiddleware(options, collectionName, op) {
+function createRequestForMiddleware(fields, options, collectionName, op) {
+  // When we're creating a request for submitting an op, let downstream middleware know.
+  if (fields && fields.$submit === true) {
+    if (!options) {
+      options = {};
+    }
+    options.isForSubmit = true;
```

Comment:
@alecgibson - Thoughts on the name here? We can pick whatever name we want, it's just a signal to the sharedb-mongo middleware handler that the getSnapshot middleware is triggering on a SubmitRequest, as opposed to a fetch.

@dwinrick-lever - Add some documentation about this new flag to the README, please!

Reply:
Feels a bit niche? Maybe something like `options.trigger = 'submitRequest'` would be more extensible? Might be a bit confusing that it's not set in other cases, but I think it's potentially worth that confusion for the extensibility.

Reply:
The alternative would be to just include `fields` in the request? Then middleware consumers can do what they like in response to `fields.$submit` (although maybe it's bad to rely on a prefixed property?).

Reply:
@alecgibson  The problem is `fields.$submit` in the first place. We shouldn't be doing that. Instead, either the caller should be populating fields based on what it knows, or we should have passed this as a different option. So, I was hesitant to let the `$submit` hack propagate into our middleware. We should plan to change this mechanism so that fields only contains actual projections and not flags. :)

Reply:
I like defining the "trigger" in options, will see if there's a generic way to have it always populated.

Reply:
@alecgibson @ericyhwang Changed to `triggeredBy = submitRequest`. Did not update docs since we haven't fully settled on this yet.

Thoughts on having sharedb specify this value? It looks like potentially we'd have the enum values: `submitRequest, queryEmit, fetch`